nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-security/src/test/groovy/org/apache/nifi/framework/security/util/SslContextFactoryTest.groovy

```diff
@@ -0,0 +1,172 @@
+
+@RunWith(JUnit4.class)
+class SslContextFactoryGroovyTest extends GroovyTestCase {
+    private static final Logger logger = LoggerFactory.getLogger(SslContextFactoryGroovyTest.class)
+    static KeyStore keyStore;
+    static KeyStore trustStore;
+    static String keystorePass = "passwordpassword"
+    static String trustStorePass = "passwordpassword"
+    static final File ksFile = new File(SslContextFactoryTest.class.getResource("/keystore.jks").toURI());
+    static final File trustFile = new File(SslContextFactoryTest.class.getResource("/truststore.jks").toURI());
+
+    @BeforeClass
+    static void setUpOnce() throws Exception {
+        Security.addProvider(new BouncyCastleProvider())
+
+        logger.metaClass.methodMissing = { String name, args ->
+            logger.info("[${name?.toUpperCase()}] ${(args as List).join(" ")}")
+        }
+
+        // prepare the keystore
+        keyStore = KeyStoreUtils.getKeyStore("JKS");
+        FileInputStream keyStoreStream = new FileInputStream(ksFile);
+        try {
+            keyStore.load(keyStoreStream, keystorePass.getChars());
+        } finally {
+            FileUtils.closeQuietly(keyStoreStream);
+        }
+
+        trustStore = KeyStoreUtils.getKeyStore("JKS");
+        FileInputStream trustStoreStream = new FileInputStream(trustFile);
+        try {
+            trustStore.load(trustStoreStream, trustStorePass.getChars());
+        } finally {
+            FileUtils.closeQuietly(trustStoreStream);
+        }
+    }
+
+    @AfterClass
+    static void tearDownOnce() throws Exception {
+        keyStore = null
+        trustStore = null
+    }
+
+    @Before
+    void setUp() throws Exception {
+
+    }
+
+    @After
+    void tearDown() throws Exception {
+        //Security.setProperty("ocsp.enable", "")
+        Security.properties.remove("ocsp.enable")
+    }
+
+    @Test
+    void testGetTrustManagerFactoryOCSPEnabled() {
+        // Act
+        TrustManagerFactory fac = SslContextFactory.getTrustManagerFactory(trustStore, true)
+
+        // Assert
+        assert fac.getTrustManagers().first().pkixParams.revocationEnabled
+        assert Security.getProperty("ocsp.enable")
+        Security.props.remove("ocsp.enable")
+    }
+
+    @Test
+    void testGetTrustManagerFactoryOCSPDisabled() {
+        // Arrange
+
+        // Act
+        TrustManagerFactory fac = SslContextFactory.getTrustManagerFactory(trustStore, false)
+
+        // Assert
+        assert fac
+        assertNull fac.getTrustManagers().first().pkixParams
+        assertNull Security.getProperty("ocsp.enable")
+    }
+
+    @Test
+    void testCreateSslContextOCSPEnabled() {
+
+        // Arrange
+        Properties nifiProps = new Properties();
+        nifiProps.setProperty(NiFiProperties.SECURITY_OCSP_ENABLED, "true")
+        nifiProps.setProperty(NiFiProperties.SECURITY_KEYSTORE, ksFile.getAbsolutePath())
+        nifiProps.setProperty(NiFiProperties.WEB_HTTP_HOST, "localhost")
```

Comment:
Neither the `WEB_HTTP_PORT` nor `WEB_HTTPS_PORT` is set here, which are the properties examined to determine if "HTTPS is enabled". An arbitrary trust manager can be created without the port (or even HTTPS enabled for the application), but I think it makes sense to include the property in this definition.